devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,550 @@
+		fingerprint manufacturer: "Viconics", model: "254-162", deviceJoinName: "Viconics Room Controller", mnmn: "SmartThings", vid: "SmartThings-smartthings-Viconics_Schneider_Room_Controller"
+		//fingerprint profileId: "0104", inClusters: "0000,0003,0201,0204,0405", outClusters: "0402,0405", manufacturer: "Viconics", model: "254-162", deviceJoinName: "VT8650xx"
+
+		// Schneider Electric SE8350 Low Voltage Fan Coil Unit (FCU) and Zone Control
+		// Raw Description 0A 0104 0301 00 0A 0201 0202 0405 0402 0406 0204 0000 0004 0003 0005 0B 0201 0202 0405 0402 0406 0204 0000 0004 0003 0005 0500
+		fingerprint manufacturer: "Schneider Electric", model: "254-145", deviceJoinName: "Schneider Electric Room Controller", vid: "SmartThings-smartthings-Viconics_Schneider_Room_Controller_Fan"
+
+		// Schneider Electric SE8650 Roof Top Unit Controller
+		// Raw Description 0A 0104 0301 00 09 0201 0405 0402 0406 0204 0000 0004 0003 0005 0A 0201 0405 0402 0406 0204 0000 0004 0003 0005 0500
+		fingerprint manufacturer: "Schneider Electric", model: "254-163", deviceJoinName: "Schneider Electric Room Controller", vid: "SmartThings-smartthings-Viconics_Schneider_Room_Controller"
+	}
+}
+
+private getTHERMOSTAT_CLUSTER() { 0x0201 }
+private getTHERMOSTAT_UI_CONFIGURATION_CLUSTER() { 0x0204 }
+private getRELATIVE_HUMIDITY_CLUSTER() {0x405}
+private getRELATIVE_HUMIDITY_MEASURED_VALUE() {0x0000}
+private getTEMPERATURE_DISPLAY_MODE() {0x0000}
+private getLOCAL_TEMPERATURE() {0x0000}
+private getCOOLING_SETPOINT() { 0x0011 }
+private getHEATING_SETPOINT() { 0x0012 }
+private getCOOLING_SETPOINT_UNOCCUPIED() { 0x0013 }
+private getHEATING_SETPOINT_UNOCCUPIED() { 0x0014 }
+private getOCCUPANCY() { 0x002 }
+private getCUSTOM_OCCUPANCY() {0x0650}
+private getCUSTOM_EFFECTIVE_OCCUPANCY() { 0x0c50 }
+private getCUSTOM_HUMIDITY() { 0x07a6 }
+private getCUSTOM_THERMOSTAT_MODE() { 0x0687 }
+private getCUSTOM_FAN_SPEED() { 0x0688 }
+private getCUSTOM_FAN_MODE() { 0x0698 }
+private getCUSTOM_THERMOSTAT_OPERATING_STATE() { 0x06BF }
+private getUNOCCUPIED_SETPOINT_CHILD_DEVICE_ID() {1}
+private getTHERMOSTAT_MODE_OFF() { 0x00 }
+private getTHERMOSTAT_MODE_AUTO() { 0x01 }
+private getTHERMOSTAT_MODE_COOL() { 0x02 }
+private getTHERMOSTAT_MODE_HEAT() { 0x03 }
+private getCUSTOM_FAN_MODE_ON() { 0x00 }
+private getCUSTOM_FAN_MODE_AUTO() { 0x01 }
+private getCUSTOM_FAN_MODE_CIRCULATE() { 0x02 }
+
+
+private getFAN_MODE_MAP() {
+	[
+			"00":"on",
+			"01":"auto",
+			"02":"circulate"
+	]
+}
+
+private getTHERMOSTAT_MODE_MAP() {
+	[
+			"00":"off",
+			"01":"auto",
+			"02":"cool",
+			"03":"heat",
+			"04":"heat"
+	]
+}
+
+private getTHERMOSTAT_OPERATING_STATE_MAP() {
+	[
+			"00":"idle",
+			"01":"cooling",
+			"02":"heating"
+	]
+}
+
+private getEFFECTIVE_OCCUPANCY_MAP() {
+	[
+			"00":"Occupied",
+			"01":"Unoccupied",
+			"02":"Override",
+			"03":"Standby"
```

Comment:
These last two are not legal values for the capability.

Reply:
fixed